### ISSUE TITLE
The Istio primary cluster VirtualService cannot be updated during canary deployment

### ISSUE THREAD

**Issue by gyz2009** [argoproj/argo-rollouts#1410]
## Summary 

What happened/what you expected to happen?
The Canary deployment has not updated the primary cluster VirtualService, the primary cluster secret has been created according to the document.
![image](https://user-images.githubusercontent.com/30786213/128999910-70a35497-26f5-4209-bb22-ad11e6f0be6e.png)

## Diagnostics

What version of Argo Rollouts are you running?
argo-rollouts:   1.0.4
istio:  1.6.14
```
# Paste the logs from the rollout controller

# Logs for the entire controller:
kubectl logs -n argo-rollouts deployment/argo-rollouts


time="2021-08-11T08:47:10Z" level=info msg="Argo Rollouts starting" version=v1.0.4+c63b6c1
time="2021-08-11T08:47:10Z" level=info msg="Creating event broadcaster"
time="2021-08-11T08:47:10Z" level=info msg="Setting up event handlers"
time="2021-08-11T08:47:10Z" level=info msg="Setting up experiments event handlers"
time="2021-08-11T08:47:10Z" level=info msg="Setting up analysis event handlers"
time="2021-08-11T08:47:10Z" level=info msg="Waiting for controller's informer caches to sync"
time="2021-08-11T08:47:10Z" level=info msg="Enqueueing parent of sample/helloworld-v0-74c489d564: Rollout sample/helloworld-v0"
time="2021-08-11T08:47:10Z" level=info msg="Enqueueing parent of sample/helloworld-v0-65fb49b99c: Rollout sample/helloworld-v0"
time="2021-08-11T08:47:10Z" level=info msg="Enqueueing parent of sample/helloworld-v0-757f74dfb9: Rollout sample/helloworld-v0"
time="2021-08-11T08:47:10Z" level=info msg="Enqueueing parent of sample/helloworld-v0-7597c8dcc6: Rollout sample/helloworld-v0"
time="2021-08-11T08:47:10Z" level=info msg="Enqueueing parent of sample/helloworld-v0-67f4597db7: Rollout sample/helloworld-v0"
time="2021-08-11T08:47:10Z" level=info msg="Enqueueing parent of sample/helloworld-v0-759b7698b8: Rollout sample/helloworld-v0"
time="2021-08-11T08:47:10Z" level=info msg="Enqueueing parent of sample/helloworld-v0-c444887b7: Rollout sample/helloworld-v0"
time="2021-08-11T08:47:10Z" level=info msg="Enqueueing parent of sample/helloworld-v0-56f6d645f6: Rollout sample/helloworld-v0"
time="2021-08-11T08:47:10Z" level=info msg="Enqueueing parent of sample/helloworld-v0-5dd856f977: Rollout sample/helloworld-v0"
time="2021-08-11T08:47:10Z" level=info msg="Enqueueing parent of sample/helloworld-v0-75d6f56d66: Rollout sample/helloworld-v0"
time="2021-08-11T08:47:10Z" level=info msg="Enqueueing parent of sample/helloworld-v0-d6c85967f: Rollout sample/helloworld-v0"
time="2021-08-11T08:47:10Z" level=info msg="Starting Controllers"
time="2021-08-11T08:47:10Z" level=info msg="Started controller"
time="2021-08-11T08:47:10Z" level=info msg="Starting Metric Server at 0.0.0.0:8090"
time="2021-08-11T08:47:10Z" level=info msg="Starting Service workers"
time="2021-08-11T08:47:10Z" level=info msg="Started Service workers"
time="2021-08-11T08:47:10Z" level=info msg="Starting Experiment workers"
time="2021-08-11T08:47:10Z" level=info msg="Started Experiment workers"
time="2021-08-11T08:47:10Z" level=info msg="Starting analysis workers"
time="2021-08-11T08:47:10Z" level=info msg="Started 30 analysis workers"
time="2021-08-11T08:47:10Z" level=info msg="Starting Ingress workers"
time="2021-08-11T08:47:10Z" level=info msg="Started Ingress workers"
time="2021-08-11T08:47:10Z" level=info msg="Starting Rollout workers"
time="2021-08-11T08:47:10Z" level=info msg="Started Rollout workers"
time="2021-08-11T08:47:10Z" level=info msg="Istio detected"
time="2021-08-11T08:47:10Z" level=info msg="Istio workers (10) started"
time="2021-08-11T08:47:10Z" level=info msg="Started syncing rollout" generation=97 namespace=sample resourceVersion=456536104 rollout=helloworld-v0
time="2021-08-11T08:47:10Z" level=info msg="Reconciling TrafficRouting with type 'Istio'" namespace=sample rollout=helloworld-v0

```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

**Comment by huikang** [comment]
@shakti-das , is this related to your PR?

**Comment by jessesuen** [comment]
Support for primary clusters in istio is not available until v1.1

**Comment by gyz2009** [comment]
@jessesuen  I tested it in 1.1.0-RC1 and it still didn't work.
```yaml
strategy:
  canary: 
    dynamicStableScale: true
    maxSurge: "25%"
    maxUnavailable: "25%"
    trafficRouting:
      istio:
        virtualService:
          name: helloworld-vsvc
          routes:
            - primary
        destinationRule:
          name: helloworld 
          canarySubsetName: canary
          stableSubsetName: stable
    steps
    - setWeight: 5
    - pause: {}
```

# Logs for the entire controller:
kubectl logs -n argo-rollouts deployment/argo-rollouts
```
time="2021-09-28T09:57:10Z" level=info msg="Patched: {\"status\":{\"conditions\":[{\"lastTransitionTime\":\"2021-09-28T03:26:59Z\",\"lastUpdateTime\":\"2021-09-28T03:26:59Z\",\"message\":\"Rollout has minimum availability\",\"reason\":\"AvailableReason\",\"status\":\"True\",\"type\":\"Available\"},{\"lastTransitionTime\":\"2021-09-28T09:28:25Z\",\"lastUpdateTime\":\"2021-09-28T09:28:25Z\",\"message\":\"Rollout is paused\",\"reason\":\"RolloutPaused\",\"status\":\"False\",\"type\":\"Paused\"},{\"lastTransitionTime\":\"2021-09-28T09:57:09Z\",\"lastUpdateTime\":\"2021-09-28T09:57:09Z\",\"message\":\"RolloutCompleted\",\"reason\":\"RolloutCompleted\",\"status\":\"False\",\"type\":\"Completed\"},{\"lastTransitionTime\":\"2021-09-28T09:28:25Z\",\"lastUpdateTime\":\"2021-09-28T09:57:09Z\",\"message\":\"Rollout \\\"helloworld-v0\\\" is progressing.\",\"reason\":\"ReplicaSetUpdated\",\"status\":\"True\",\"type\":\"Progressing\"}],\"currentPodHash\":\"785b465f9b\",\"currentStepIndex\":0,\"message\":\"more replicas need to be updated\",\"observedGeneration\":\"184\",\"phase\":\"Progressing\",\"updatedReplicas\":null}}" generation=184 namespace=sample resourceVersion=552444764 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="persisted to informer" generation=184 namespace=sample resourceVersion=552444765 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Reconciliation completed" generation=184 namespace=sample resourceVersion=552444764 rollout=helloworld-v0 time_ms=39.379401
time="2021-09-28T09:57:10Z" level=info msg="Start processing" resource=sample/helloworld-v0
time="2021-09-28T09:57:10Z" level=error msg="Failed to process: secret \"argo-rollouts-notification-secret\" not found" resource=sample/helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Started syncing rollout" generation=184 namespace=sample resourceVersion=552444765 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Updating replica set 'helloworld-v0-785b465f9b' revision from 0 to 66" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Created ReplicaSet helloworld-v0-785b465f9b" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Enqueueing parent of sample/helloworld-v0-785b465f9b: Rollout sample/helloworld-v0"
time="2021-09-28T09:57:10Z" level=info msg="Enqueueing parent of sample/helloworld-v0-785b465f9b: Rollout sample/helloworld-v0"
time="2021-09-28T09:57:10Z" level=info msg="Rollout updated to revision 66" event_reason=RolloutUpdated namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Created ReplicaSet helloworld-v0-785b465f9b (revision 66)" event_reason=NewReplicaSetCreated namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Event(v1.ObjectReference{Kind:\"Rollout\", Namespace:\"sample\", Name:\"helloworld-v0\", UID:\"a7ba239b-035f-49e7-8a1f-f90b8ba8182b\", APIVersion:\"argoproj.io/v1alpha1\", ResourceVersion:\"552444771\", FieldPath:\"\"}): type: 'Normal' reason: 'RolloutUpdated' Rollout updated to revision 66"
time="2021-09-28T09:57:10Z" level=info msg="Event(v1.ObjectReference{Kind:\"Rollout\", Namespace:\"sample\", Name:\"helloworld-v0\", UID:\"a7ba239b-035f-49e7-8a1f-f90b8ba8182b\", APIVersion:\"argoproj.io/v1alpha1\", ResourceVersion:\"552444771\", FieldPath:\"\"}): type: 'Normal' reason: 'NewReplicaSetCreated' Created ReplicaSet helloworld-v0-785b465f9b (revision 66)"
time="2021-09-28T09:57:10Z" level=info msg="Start processing" resource=sample/helloworld-v0
time="2021-09-28T09:57:10Z" level=error msg="Failed to process: secret \"argo-rollouts-notification-secret\" not found" resource=sample/helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Set rollout condition: &RolloutCondition{Type:Progressing,Status:True,LastUpdateTime:2021-09-28 09:57:10.079499225 +0000 UTC m=+186313.712012842,LastTransitionTime:2021-09-28 09:57:10.079499312 +0000 UTC m=+186313.712012931,Reason:NewReplicaSetCreated,Message:Created new replica set \"helloworld-v0-785b465f9b\",}" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Reconciling TrafficRouting with type 'Istio'" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Previous weights: &TrafficWeights{Canary:WeightDestination{Weight:0,ServiceName:helloworld-canary,PodTemplateHash:788f49f4f,},Stable:WeightDestination{Weight:100,ServiceName:helloworld,PodTemplateHash:788f49f4f,},Additional:[]WeightDestination{},Verified:nil,}" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="New weights: &TrafficWeights{Canary:WeightDestination{Weight:0,ServiceName:helloworld-canary,PodTemplateHash:785b465f9b,},Stable:WeightDestination{Weight:100,ServiceName:helloworld,PodTemplateHash:788f49f4f,},Additional:[]WeightDestination{},Verified:nil,}" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Start processing" resource=sample/helloworld-v0
time="2021-09-28T09:57:10Z" level=error msg="Failed to process: secret \"argo-rollouts-notification-secret\" not found" resource=sample/helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="persisted to informer" generation=185 namespace=sample resourceVersion=552444773 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=error msg="roCtx.reconcile err Operation cannot be fulfilled on replicasets.apps \"helloworld-v0-785b465f9b\": the object has been modified; please apply your changes to the latest version and try again" generation=184 namespace=sample resourceVersion=552444765 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Reconciliation completed" generation=184 namespace=sample resourceVersion=552444765 rollout=helloworld-v0 time_ms=82.942026
time="2021-09-28T09:57:10Z" level=error msg="rollout syncHandler error: Operation cannot be fulfilled on replicasets.apps \"helloworld-v0-785b465f9b\": the object has been modified; please apply your changes to the latest version and try again" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="rollout syncHandler queue retries: 6 : key \"sample/helloworld-v0\"" namespace=sample rollout=helloworld-v0
E0928 09:57:10.116044       1 controller.go:174] Operation cannot be fulfilled on replicasets.apps "helloworld-v0-785b465f9b": the object has been modified; please apply your changes to the latest version and try again
time="2021-09-28T09:57:10Z" level=info msg="Started syncing rollout" generation=185 namespace=sample resourceVersion=552444773 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Reconciling TrafficRouting with type 'Istio'" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Previous weights: &TrafficWeights{Canary:WeightDestination{Weight:0,ServiceName:helloworld-canary,PodTemplateHash:788f49f4f,},Stable:WeightDestination{Weight:100,ServiceName:helloworld,PodTemplateHash:788f49f4f,},Additional:[]WeightDestination{},Verified:nil,}" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="New weights: &TrafficWeights{Canary:WeightDestination{Weight:0,ServiceName:helloworld-canary,PodTemplateHash:785b465f9b,},Stable:WeightDestination{Weight:100,ServiceName:helloworld,PodTemplateHash:788f49f4f,},Additional:[]WeightDestination{},Verified:nil,}" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Scaled up ReplicaSet helloworld-v0-785b465f9b (revision 66) from 0 to 1" event_reason=ScalingReplicaSet namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Not finished reconciling new ReplicaSet 'helloworld-v0-785b465f9b'" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Not finished reconciling ReplicaSets" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Event(v1.ObjectReference{Kind:\"Rollout\", Namespace:\"sample\", Name:\"helloworld-v0\", UID:\"a7ba239b-035f-49e7-8a1f-f90b8ba8182b\", APIVersion:\"argoproj.io/v1alpha1\", ResourceVersion:\"552444773\", FieldPath:\"\"}): type: 'Normal' reason: 'ScalingReplicaSet' Scaled up ReplicaSet helloworld-v0-785b465f9b (revision 66) from 0 to 1"
time="2021-09-28T09:57:10Z" level=info msg="Timed out (false) [last progress check: 2021-09-28 09:57:10 +0000 UTC - now: 2021-09-28 09:57:10.131747563 +0000 UTC m=+186313.764261180]" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Enqueueing parent of sample/helloworld-v0-785b465f9b: Rollout sample/helloworld-v0"
time="2021-09-28T09:57:10Z" level=info msg="Patched: {\"status\":{\"canary\":{\"weights\":{\"canary\":{\"podTemplateHash\":\"785b465f9b\"}}},\"observedGeneration\":\"185\"}}" generation=185 namespace=sample resourceVersion=552444773 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="persisted to informer" generation=185 namespace=sample resourceVersion=552444775 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Reconciliation completed" generation=185 namespace=sample resourceVersion=552444773 rollout=helloworld-v0 time_ms=38.335827
time="2021-09-28T09:57:10Z" level=info msg="Start processing" resource=sample/helloworld-v0
time="2021-09-28T09:57:10Z" level=error msg="Failed to process: secret \"argo-rollouts-notification-secret\" not found" resource=sample/helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Started syncing rollout" generation=185 namespace=sample resourceVersion=552444775 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Reconciling TrafficRouting with type 'Istio'" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Timed out (false) [last progress check: 2021-09-28 09:57:10 +0000 UTC - now: 2021-09-28 09:57:10.160262469 +0000 UTC m=+186313.792776081]" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="No status changes. Skipping patch" generation=185 namespace=sample resourceVersion=552444775 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Queueing up rollout for a progress after 599s" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Reconciliation completed" generation=185 namespace=sample resourceVersion=552444775 rollout=helloworld-v0 time_ms=6.099108
time="2021-09-28T09:57:10Z" level=info msg="Started syncing rollout" generation=185 namespace=sample resourceVersion=552444775 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Reconciling TrafficRouting with type 'Istio'" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Timed out (false) [last progress check: 2021-09-28 09:57:10 +0000 UTC - now: 2021-09-28 09:57:10.162582816 +0000 UTC m=+186313.795096427]" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="No status changes. Skipping patch" generation=185 namespace=sample resourceVersion=552444775 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Queueing up rollout for a progress after 599s" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Reconciliation completed" generation=185 namespace=sample resourceVersion=552444775 rollout=helloworld-v0 time_ms=2.284686
time="2021-09-28T09:57:10Z" level=info msg="Enqueueing parent of sample/helloworld-v0-785b465f9b: Rollout sample/helloworld-v0"
time="2021-09-28T09:57:10Z" level=info msg="Started syncing rollout" generation=185 namespace=sample resourceVersion=552444775 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Reconciling TrafficRouting with type 'Istio'" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Timed out (false) [last progress check: 2021-09-28 09:57:10 +0000 UTC - now: 2021-09-28 09:57:10.193262166 +0000 UTC m=+186313.825775794]" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="No status changes. Skipping patch" generation=185 namespace=sample resourceVersion=552444775 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Queueing up rollout for a progress after 599s" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Reconciliation completed" generation=185 namespace=sample resourceVersion=552444775 rollout=helloworld-v0 time_ms=3.53157
time="2021-09-28T09:57:10Z" level=info msg="Enqueueing parent of sample/helloworld-v0-785b465f9b: Rollout sample/helloworld-v0"
time="2021-09-28T09:57:10Z" level=info msg="Started syncing rollout" generation=185 namespace=sample resourceVersion=552444775 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Reconciling TrafficRouting with type 'Istio'" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Patched: {\"status\":{\"HPAReplicas\":13,\"conditions\":[{\"lastTransitionTime\":\"2021-09-28T03:26:59Z\",\"lastUpdateTime\":\"2021-09-28T03:26:59Z\",\"message\":\"Rollout has minimum availability\",\"reason\":\"AvailableReason\",\"status\":\"True\",\"type\":\"Available\"},{\"lastTransitionTime\":\"2021-09-28T09:28:25Z\",\"lastUpdateTime\":\"2021-09-28T09:28:25Z\",\"message\":\"Rollout is paused\",\"reason\":\"RolloutPaused\",\"status\":\"False\",\"type\":\"Paused\"},{\"lastTransitionTime\":\"2021-09-28T09:57:09Z\",\"lastUpdateTime\":\"2021-09-28T09:57:09Z\",\"message\":\"RolloutCompleted\",\"reason\":\"RolloutCompleted\",\"status\":\"False\",\"type\":\"Completed\"},{\"lastTransitionTime\":\"2021-09-28T09:28:25Z\",\"lastUpdateTime\":\"2021-09-28T09:57:10Z\",\"message\":\"ReplicaSet \\\"helloworld-v0-785b465f9b\\\" is progressing.\",\"reason\":\"ReplicaSetUpdated\",\"status\":\"True\",\"type\":\"Progressing\"}],\"replicas\":13,\"updatedReplicas\":1}}" generation=185 namespace=sample resourceVersion=552444775 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="persisted to informer" generation=185 namespace=sample resourceVersion=552444782 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Reconciliation completed" generation=185 namespace=sample resourceVersion=552444775 rollout=helloworld-v0 time_ms=22.012904
time="2021-09-28T09:57:10Z" level=info msg="Start processing" resource=sample/helloworld-v0
time="2021-09-28T09:57:10Z" level=error msg="Failed to process: secret \"argo-rollouts-notification-secret\" not found" resource=sample/helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Started syncing rollout" generation=185 namespace=sample resourceVersion=552444782 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Reconciling TrafficRouting with type 'Istio'" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Timed out (false) [last progress check: 2021-09-28 09:57:10 +0000 UTC - now: 2021-09-28 09:57:10.235730978 +0000 UTC m=+186313.868244588]" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="No status changes. Skipping patch" generation=185 namespace=sample resourceVersion=552444782 rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Queueing up rollout for a progress after 599s" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:10Z" level=info msg="Reconciliation completed" generation=185 namespace=sample resourceVersion=552444782 rollout=helloworld-v0 time_ms=2.340827
time="2021-09-28T09:57:41Z" level=info msg="Enqueueing parent of sample/helloworld-v0-785b465f9b: Rollout sample/helloworld-v0"
time="2021-09-28T09:57:41Z" level=info msg="Started syncing rollout" generation=185 namespace=sample resourceVersion=552444782 rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Switched selector for service 'helloworld-canary' from '788f49f4f' to '785b465f9b'" event_reason=SwitchService namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Reconciling TrafficRouting with type 'Istio'" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="syncing service" namespace=sample rollout=helloworld-v0 service=helloworld-canary
time="2021-09-28T09:57:41Z" level=info msg="Event(v1.ObjectReference{Kind:\"Rollout\", Namespace:\"sample\", Name:\"helloworld-v0\", UID:\"a7ba239b-035f-49e7-8a1f-f90b8ba8182b\", APIVersion:\"argoproj.io/v1alpha1\", ResourceVersion:\"552444782\", FieldPath:\"\"}): type: 'Normal' reason: 'SwitchService' Switched selector for service 'helloworld-canary' from '788f49f4f' to '785b465f9b'"
time="2021-09-28T09:57:41Z" level=info msg="Previous weights: &TrafficWeights{Canary:WeightDestination{Weight:0,ServiceName:helloworld-canary,PodTemplateHash:785b465f9b,},Stable:WeightDestination{Weight:100,ServiceName:helloworld,PodTemplateHash:788f49f4f,},Additional:[]WeightDestination{},Verified:nil,}" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="New weights: &TrafficWeights{Canary:WeightDestination{Weight:5,ServiceName:helloworld-canary,PodTemplateHash:785b465f9b,},Stable:WeightDestination{Weight:95,ServiceName:helloworld,PodTemplateHash:788f49f4f,},Additional:[]WeightDestination{},Verified:nil,}" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Rollout step 1/2 completed (setWeight: 5)" event_reason=RolloutStepCompleted namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Event(v1.ObjectReference{Kind:\"Rollout\", Namespace:\"sample\", Name:\"helloworld-v0\", UID:\"a7ba239b-035f-49e7-8a1f-f90b8ba8182b\", APIVersion:\"argoproj.io/v1alpha1\", ResourceVersion:\"552444782\", FieldPath:\"\"}): type: 'Normal' reason: 'RolloutStepCompleted' Rollout step 1/2 completed (setWeight: 5)"
time="2021-09-28T09:57:41Z" level=info msg="Patched: {\"status\":{\"availableReplicas\":13,\"canary\":{\"weights\":{\"canary\":{\"weight\":5},\"stable\":{\"weight\":95}}},\"conditions\":[{\"lastTransitionTime\":\"2021-09-28T03:26:59Z\",\"lastUpdateTime\":\"2021-09-28T03:26:59Z\",\"message\":\"Rollout has minimum availability\",\"reason\":\"AvailableReason\",\"status\":\"True\",\"type\":\"Available\"},{\"lastTransitionTime\":\"2021-09-28T09:28:25Z\",\"lastUpdateTime\":\"2021-09-28T09:28:25Z\",\"message\":\"Rollout is paused\",\"reason\":\"RolloutPaused\",\"status\":\"False\",\"type\":\"Paused\"},{\"lastTransitionTime\":\"2021-09-28T09:57:09Z\",\"lastUpdateTime\":\"2021-09-28T09:57:09Z\",\"message\":\"RolloutCompleted\",\"reason\":\"RolloutCompleted\",\"status\":\"False\",\"type\":\"Completed\"},{\"lastTransitionTime\":\"2021-09-28T09:28:25Z\",\"lastUpdateTime\":\"2021-09-28T09:57:41Z\",\"message\":\"ReplicaSet \\\"helloworld-v0-785b465f9b\\\" is progressing.\",\"reason\":\"ReplicaSetUpdated\",\"status\":\"True\",\"type\":\"Progressing\"}],\"currentStepIndex\":1,\"readyReplicas\":13}}" generation=185 namespace=sample resourceVersion=552444782 rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="persisted to informer" generation=185 namespace=sample resourceVersion=552445843 rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Reconciliation completed" generation=185 namespace=sample resourceVersion=552444782 rollout=helloworld-v0 time_ms=60.127796
time="2021-09-28T09:57:41Z" level=info msg="Start processing" resource=sample/helloworld-v0
time="2021-09-28T09:57:41Z" level=error msg="Failed to process: secret \"argo-rollouts-notification-secret\" not found" resource=sample/helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Started syncing rollout" generation=185 namespace=sample resourceVersion=552445843 rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Reconciling TrafficRouting with type 'Istio'" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Reconciling canary pause step (stepIndex: 1/2)" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Not finished reconciling Canary Pause" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Adding pause reason CanaryPauseStep with start time 2021-09-28T09:57:41Z" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Rollout is paused (CanaryPauseStep)" event_reason=RolloutPaused namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Patched: {\"status\":{\"controllerPause\":true,\"message\":\"CanaryPauseStep\",\"pauseConditions\":[{\"reason\":\"CanaryPauseStep\",\"startTime\":\"2021-09-28T09:57:41Z\"}],\"phase\":\"Paused\"}}" generation=185 namespace=sample resourceVersion=552445843 rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Event(v1.ObjectReference{Kind:\"Rollout\", Namespace:\"sample\", Name:\"helloworld-v0\", UID:\"a7ba239b-035f-49e7-8a1f-f90b8ba8182b\", APIVersion:\"argoproj.io/v1alpha1\", ResourceVersion:\"552445843\", FieldPath:\"\"}): type: 'Normal' reason: 'RolloutPaused' Rollout is paused (CanaryPauseStep)"
time="2021-09-28T09:57:41Z" level=info msg="persisted to informer" generation=185 namespace=sample resourceVersion=552445845 rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Reconciliation completed" generation=185 namespace=sample resourceVersion=552445843 rollout=helloworld-v0 time_ms=39.378617999999996
time="2021-09-28T09:57:41Z" level=info msg="Start processing" resource=sample/helloworld-v0
time="2021-09-28T09:57:41Z" level=error msg="Failed to process: secret \"argo-rollouts-notification-secret\" not found" resource=sample/helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Started syncing rollout" generation=185 namespace=sample resourceVersion=552445845 rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Patched conditions: {\"status\":{\"conditions\":[{\"lastTransitionTime\":\"2021-09-28T03:26:59Z\",\"lastUpdateTime\":\"2021-09-28T03:26:59Z\",\"message\":\"Rollout has minimum availability\",\"reason\":\"AvailableReason\",\"status\":\"True\",\"type\":\"Available\"},{\"lastTransitionTime\":\"2021-09-28T09:57:09Z\",\"lastUpdateTime\":\"2021-09-28T09:57:09Z\",\"message\":\"RolloutCompleted\",\"reason\":\"RolloutCompleted\",\"status\":\"False\",\"type\":\"Completed\"},{\"lastTransitionTime\":\"2021-09-28T09:57:41Z\",\"lastUpdateTime\":\"2021-09-28T09:57:41Z\",\"message\":\"Rollout is paused\",\"reason\":\"RolloutPaused\",\"status\":\"Unknown\",\"type\":\"Progressing\"},{\"lastTransitionTime\":\"2021-09-28T09:57:41Z\",\"lastUpdateTime\":\"2021-09-28T09:57:41Z\",\"message\":\"Rollout is paused\",\"reason\":\"RolloutPaused\",\"status\":\"True\",\"type\":\"Paused\"}]}}" generation=185 namespace=sample resourceVersion=552445845 rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Reconciling TrafficRouting with type 'Istio'" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Start processing" resource=sample/helloworld-v0
time="2021-09-28T09:57:41Z" level=error msg="Failed to process: secret \"argo-rollouts-notification-secret\" not found" resource=sample/helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Reconciling canary pause step (stepIndex: 1/2)" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Not finished reconciling Canary Pause" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="No status changes. Skipping patch" generation=185 namespace=sample resourceVersion=552445845 rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="persisted to informer" generation=185 namespace=sample resourceVersion=552445847 rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Reconciliation completed" generation=185 namespace=sample resourceVersion=552445845 rollout=helloworld-v0 time_ms=41.413268
time="2021-09-28T09:57:41Z" level=info msg="Started syncing rollout" generation=185 namespace=sample resourceVersion=552445847 rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Reconciling TrafficRouting with type 'Istio'" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Reconciling canary pause step (stepIndex: 1/2)" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Not finished reconciling Canary Pause" namespace=sample rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="No status changes. Skipping patch" generation=185 namespace=sample resourceVersion=552445847 rollout=helloworld-v0
time="2021-09-28T09:57:41Z" level=info msg="Reconciliation completed" generation=185 namespace=sample resourceVersion=552445847 rollout=helloworld-v0 time_ms=12.117034
```

![image](https://user-images.githubusercontent.com/30786213/135070276-469d7fcc-771f-4016-b9fd-e71754dacc6d.png)
